pkg/kubelet/kubelet.go

```diff
@@ -2434,6 +2434,7 @@ func (kl *Kubelet) HandlePodDeletions(pods []*api.Pod) {
 			glog.V(2).Infof("Failed to delete pod %q, err: %v", format.Pod(pod), err)
 		}
 		kl.probeManager.RemovePod(pod)
+		kl.statusManager.DeletePodStatus(pod.UID)
```

Comment:
Yes, there are a lot of races in cleanup. I think this will be eventually handled by the periodic cleanup routine, similar to other things.